Task6/Catalog.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Task6
+{
+	[Serializable]
+	public class Catalog 
+	{
+		public  Dictionary<string, Book> catalog;
+
+		public Catalog()
+		{
+			catalog = new Dictionary<string, Book>();
+		}
+
+		public void AddBook(Book book)
+		{
+			if (catalog.ContainsKey(FormattedIsbn(book.Isbn)))
+			{
+				throw new ArgumentException("Book is already added!");
+			}
+			catalog[book.Isbn] = book;
+		}
```

Comment:
Move isbn check to Book class, Catalog has to store only well formatted books, not logics with validating isbns